kratos/modeler/create_entities_from_geometries_modeler.cpp

```diff
@@ -0,0 +1,168 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/parallel_environment.h"
+#include "utilities/entities_utilities.h"
+#include "utilities/variable_utils.h"
+#include "create_entities_from_geometries_modeler.h"
+
+namespace Kratos
+{
+
+namespace {
+
+/**
+ * @brief Create entities from geometries.
+ * @param EntityName the name of the entity
+ * @param rModelPart the model part
+ */
+template <class TEntitiesContainerType, class TEntityIdentifier>
+void CreateEntitiesFromGeometries(
+    const TEntityIdentifier& rEntityIdentifier,
+    ModelPart& rModelPart
+    )
+{
+    // Create the entities container and allocate space
+    TEntitiesContainerType entities_to_add;
+    entities_to_add.reserve(rModelPart.NumberOfGeometries());
+
+    // Get current max element id
+    using SizeType = std::size_t;
+    SizeType max_id;
+    const auto& r_root_model_part = rModelPart.GetRootModelPart();
+    if constexpr (std::is_same<typename TEntitiesContainerType::value_type, Element>::value) {
+        max_id = block_for_each<MaxReduction<SizeType>>(r_root_model_part.Elements(), [](auto& rElement){
+            return rElement.Id();
+        });
+    } else {
```

Comment:
Check for conditions explicitly and throw error otherwise. otherwise you can instantiate this with nodes or mpc